src/main/java/nextstep/security/config/HttpSecurityConfiguration.java

```diff
@@ -0,0 +1,33 @@
+package nextstep.security.config;
+
+import nextstep.oauth2.registration.ClientRegistrationRepository;
+import nextstep.security.access.hierarchicalroles.NullRoleHierarchy;
+import nextstep.security.access.hierarchicalroles.RoleHierarchy;
+import nextstep.security.authentication.AuthenticationManager;
+import nextstep.security.builder.HttpSecurity;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Scope;
+
+@Configuration
+public class HttpSecurityConfiguration {
+
+    @Bean
+    @Scope("prototype")
+    public HttpSecurity httpSecurity(
+              AuthenticationManager authenticationManager
+            , RoleHierarchy roleHierarchy
```

Comment:
```suggestion
            , @Autowired(required = false) RoleHierarchy roleHierarchy
```

스프링 빈 주입 시 기본적으로 null이 아닌 빈을 주입하기 때문에 별도의 설정을 해줘야 겠네요!

Reply:
앗 .. 그러네요 [c5bb9ff](https://github.com/next-step/spring-security-refactoring/pull/4/commits/c5bb9ff944b4b4aa491ca17501b4695c171f9cc0) 해당커밋에 반영해두었습니다.